resources/mediaprocessor.py

```diff
@@ -1490,64 +1494,85 @@ def checkDisposition(self, allowed, source):
+        self.log.debug(f"Selected hwaccel decoder pairs: {self.settings.hwaccel_decoders}.")
+        self.log.debug(f"Supported ffmpeg hwaccels: {hwaccels}.")
+        self.log.debug(f"Input format: {pix_fmt} with bit depth {bit_depth}.")
+
+        def _add_hwaccel_opts(_hwaccel: str, _decoder: str):
+            # set hwaccel and hwaccel_output_format, if specified
+            opts.extend(['-hwaccel', _hwaccel])
+            if self.settings.hwoutputfmt.get(_hwaccel):
+                opts.extend(['-hwaccel_output_format', self.settings.hwoutputfmt[_hwaccel]])
+
+            # set hw device, if specified
+            nonlocal device
+            device = self.settings.hwdevices.get(_hwaccel)
+            if device:
+                self.log.debug("Setting hwaccel device to %s." % device)
+                opts.extend(['-init_hw_device', '%s=sma:%s' % (_hwaccel, device)])
+                opts.extend(['-hwaccel_device', 'sma'])
+
+            # Set vcodec
+            opts.extend(['-vcodec', _decoder])
+
+        # If there's a manually specified hwaccel/decoder pairing for this codec, use it.
+        if video_codec in self.settings.hwaccel_decoder_override:
+            hwaccel, target_decoder = self.settings.hwaccel_decoder_override[video_codec].split('.')
+
+            if target_decoder in codecs[video_codec]['decoders']:
+                self.log.debug(f"Detecting override for codec={video_codec}, setting hwaccel={hwaccel} and vcodec={target_decoder}. [hwaccel-decoders]")
+
+                _add_hwaccel_opts(hwaccel, target_decoder)
+
+        else:
+            """
+            Find the first hwaccel platform that:
+                1. Is specified by settings.hwaccels.
+                2. Is included in this build of ffmpeg.
+                3. Is considered by ffmpeg to be a valid decoder for the input codec.
+                4. Is included in hwaccel_decoders.
+                5. Supports the given pixel format.
+    
+            Given that all these criteria are met, opts will be extended to include -hwaccel and -vcodec. Additional
+            parameters (-hwaccel_output_format, -init_hw_device/-hwaccel_device) will be included if specified by their
+            corresponding setting.
+            """
+            for hwaccel in self.settings.hwaccels:
+                if hwaccel in hwaccels:
+                    target_decoder = self.converter.ffmpeg.hwaccel_decoder(
+                        video_codec,
+                        self.settings.hwoutputfmt.get(hwaccel, hwaccel)
+                    )
+
+                    self.log.debug(f"Target decoder: {target_decoder}")
+                    self.log.debug(f"Target decoder pixel formats: {self.converter.ffmpeg.decoder_formats(target_decoder)}.")
+
+                    is_supported_decoder = target_decoder in codecs[video_codec]['decoders']
+
+                    if is_supported_decoder and target_decoder in self.settings.hwaccel_decoders:
```

Comment:
This slightly modifies existing behavior. Only specifying `hwaccels=` in settings will now do nothing. Most examples I've seen of users attempting hardware acceleration on this repo have been specifying their decoders anyways.